log4j-core/pom.xml

```diff
@@ -66,15 +66,16 @@
       <!-- Remove `transitive` from optional dependencies -->
       com.fasterxml.jackson.annotation;transitive=false,
       com.lmax.disruptor;transitive=false,
-      com.fasterxml.jackson.core;substitute="jackson-core";transitive=false,
-      com.fasterxml.jackson.databind;substitute="jackson-databind";transitive=false,
-      com.fasterxml.jackson.dataformat.xml;substitute="jackson-dataformat-xml";transitive=false,
-      com.fasterxml.jackson.dataformat.yaml;substitute="jackson-dataformat-yaml";transitive=false,
+      com.fasterxml.jackson.core;transitive=false,
+      com.fasterxml.jackson.databind;transitive=false,
+      com.fasterxml.jackson.dataformat.xml;transitive=false,
+      com.fasterxml.jackson.dataformat.yaml;transitive=false,
       java.naming;transitive=false,
       org.apache.commons.csv;transitive=false,
       org.fusesource.jansi;transitive=false,
+      <!-- Newer versions have a module descriptor -->
+      com.conversantmedia.disruptor;substitute="disruptor";transitive=false;static=true,
```

Comment:
Why `substitute` is not necessary for Jackson et al., whereas it is for Disruptor?

Reply:
Good question!

In `2.x` we are stuck with `com.conversant:disruptor:1.2.15`, which is the last version that runs on JRE 8. In the 1.2.16 **patch** release, Conversant:

 - bumped the required JRE version to 9,
 - added a JPMS module descriptor.

Therefore I assume that all the users that use Conversant Media Disruptor and JPMS, will use version 1.2.16+.

Reply:
Got it. Jackson is still at version `2.16.0`, right? That is, no JPMS descriptor. Shouldn't we be keeping `substitute` there?

Reply:
Jackson has a JPMS descriptor in `META-INF/versions/9`. BND 6.x could not find it, but BND 7.x can.